transports/quic/tests/smoke.rs

```diff
@@ -182,6 +183,66 @@ fn concurrent_connections_and_streams_tokio() {
         .quickcheck(prop::<quic::tokio::Provider> as fn(_, _) -> _);
 }
 
+#[cfg(feature = "async-std")]
+#[async_std::test]
+async fn backpressure() {
+    let _ = env_logger::try_init();
+    let max_stream_data = quic::Config::new(&generate_tls_keypair()).max_stream_data;
+
+    let (mut stream_a, mut stream_b) = build_streams::<quic::async_std::Provider>().await;
+
+    let data = vec![0; max_stream_data as usize - 1];
+
+    stream_a.write_all(&data).await.unwrap();
+
+    let too_much_data = vec![0; 10];
+
+    assert!(
+        async_std::future::timeout(Duration::from_secs(1), stream_a.write_all(&too_much_data))
```

Comment:
Usage of a timeout in a unit test is unfortunate. That said, I can not think of an alternative, and I think it is worth it. Thus I suggest let's keep as is.

Reply:
You could assert that it returns `Poll::Pending` instead and `Poll::Ready` once you read from the other side again.

Reply:
> You could assert that it returns `Poll::Pending` instead and `Poll::Ready` once you read from the other side again.

:+1:. Done in 9f2a7ee.